apps/xmtp.chat/src/components/Messages/TransactionReference.tsx

```diff
@@ -0,0 +1,36 @@
+import { Anchor } from "@mantine/core";
+import type { TransactionReference } from "@xmtp/content-type-transaction-reference";
+import * as viemChains from "viem/chains";
+
+export type TransactionReferenceProps = {
+  content: TransactionReference;
+};
+
+export const TransactionReferenceUI: React.FC<TransactionReferenceProps> = ({
+  content,
+}) => {
+  const chains = Object.values(viemChains);
```

Comment:
this should exist outside of the rendering function as it only needs to be calculated once and doesn't depending on any of the component's props/data.

Reply:
Moved it to inside useMemo. Is this fine?